### PR TITLE
Shenli/optimizer error

### PR DESCRIPTION
**Pull request by shenli**
@coocood

### CODE REVIEW

File: optimizer/optimizer.go [hunk 1]
```diff
@@ -110,3 +130,12 @@ func IsSupported(node ast.Node) bool {
 	node.Accept(&checker)
 	return !checker.unsupported
 }
+
+func init() {
+	mySQLErrCodes := map[terror.ErrCode]uint16{
+		CodeOneColumn:     mysql.ErrBadDb,
+		CodeMultiWildCard: mysql.ErrParse,
+		CodeSameColumns:   mysql.ErrOperandColumns,
```
Comment (coocood) [inline optimizer/optimizer.go]: Missing `CodeRowColumns `

**Comment by shenli** [comment]
PTAL

**Comment by qiuyesuifeng** [comment]
LGTM

File: tidb-server/server/server_test.go [hunk 2]
```diff
@@ -238,6 +238,14 @@ func runTestErrorCode(c *C) {
 		checkErrorCode(c, err, tmysql.ErrBadDb)
 		_, err = txn2.Exec("select * from tbl_not_exists;")
 		checkErrorCode(c, err, tmysql.ErrNoSuchTable)
+
+		// Optimizer errors
+		_, err = txn2.Exec("select *, * from test;")
```
Comment (qiuyesuifeng) [inline tidb-server/server/server_test.go]: Add test for CodeRowColumns?

Comment (shenli) [inline tidb-server/server/server_test.go]: It is useless. PTAL

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**